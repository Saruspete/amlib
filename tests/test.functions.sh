
glbref="global"
glbtst="global"
ovrref="global"
ovrtst="global"
typeset typref="global"
typeset typtst="global"


function func1 {
	
	echo "++ func1: glb:$glbref typ:$typref ovr:$ovrref"
	glbtst="func1"
	typtst="func1"
	typeset ovrtst="func1"
	typeset typloc="func1"
	ovrloc="func1"
}

func2 () {
	echo "++ func2: glb:$glbref typ:$typref ovr:$ovrref"
	glbtst="func2"
	typtst="func2"
	typeset ovrtst="func2"
	typeset typloc="func2"
	ovrloc="func2"
}

# Test 1
echo "== global: glb:$glbtst typ:$typtst ovr:$ovrtst loctyp:$typloc locovr=$ovrloc"
func1
echo "== global: glb:$glbtst typ:$typtst ovr:$ovrtst loctyp:$typloc locovr=$ovrloc"
echo

# Reset
glbtst="global"
typeset typtst="global"
ovrtst="global"


echo "== global: glb:$glbtst typ:$typtst ovr:$ovrtst loctyp:$typloc locovr=$ovrloc"
func2
echo "== global: glb:$glbtst typ:$typtst ovr:$ovrtst loctyp:$typloc locovr=$ovrloc"
