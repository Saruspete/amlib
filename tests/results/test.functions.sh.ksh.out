== global: glb:global typ:global ovr:global loctyp: locovr=
++ func1: glb:global typ:global ovr:global
== global: glb:func1 typ:func1 ovr:global loctyp: locovr=func1

== global: glb:global typ:global ovr:global loctyp: locovr=func1
++ func2: glb:global typ:global ovr:global
== global: glb:func2 typ:func2 ovr:func2 loctyp:func2 locovr=func2
